app/Services/VocabularyServiceProvider.js

```diff
@@ -2,41 +2,41 @@ const { VocabularyCard, Translation } = require('../../database');
 const { Drawer } = require('../../database');
 
 // create language package
-async function createVocabularyCard({ languagePackageId, groupId }, name, userId, activate) {
+async function createVocabularyCard({ languagePackageId, groupId }, name, description, userId, activate, res) {
   // if activate = false store vocabulary card in drawer 0 directly
+
+  let drawer;
+
   if (!activate) {
-    const drawer = await Drawer.findOne({
-      attributes: ['id', 'stage'],
+    drawer = await Drawer.findOne({
+      attributes: ['id'],
       where: {
         stage: 0,
         languagePackageId,
         userId,
       },
     });
 
-    const vocabularyCard = await VocabularyCard.create({
-      userId: userId,
-      languagePackageId: languagePackageId,
-      groupId: groupId,
-      drawerId: drawer.id,
-      name: name,
-      lastQuery: new Date(),
-      active: true,
+    if (!drawer) {
+      res.status(400).end();
+      return false;
+    }
+  } else {
+    // if user directly activates card, store it in drawer 1
+    drawer = await Drawer.findOne({
+      attributes: ['id'],
+      where: {
+        stage: 1,
+        languagePackageId: languagePackageId,
+        userId: userId,
+      },
     });
 
-    return vocabularyCard;
+    if (!drawer) {
+      res.status(400).end();
+      return false;
+    }
```

Comment:
here we can set a variable for the drawer stage conditionally and use it to finde the derawer instead of copy the whole block in a if else.